src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -47,34 +47,36 @@
  * Queues up elements until either a duration of time has passed or any threshold in a given set of
  * thresholds is breached, and then delivers the elements in a bundle to the consumer.
  */
-public final class ThresholdBundler<E> {
+public final class ThresholdBundler<E extends Bundle<E>> {
 
   private ImmutableList<BundlingThreshold<E>> thresholdPrototypes;
   private final Duration maxDelay;
   private final BundlingFlowController<E> flowController;
+  private final BundleSupplier<E> bundleFactory;
```

Comment:
@garrettjonesgoogle see my comment about using Java 8 name to simplify future migration. The variable names should match regardless.